### PR TITLE
Adding a configuration section for RoutingAutoBundle

### PR DESCRIPTION
**Pull request by pcabreus**

### CODE REVIEW

File: bundles/routing_auto/configuration.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+Configuration
+-------------
+
+``adapter``
+~~~~~~~~~~~
+
+**type**: ``scalar`` **default**: ``doctrine_phpcr_odm`` if ``persistence`` configuration option is set ``phpcr``
+
+This define the adapter used to manage routes.
```
Comment (dantleech) [inline bundles/routing_auto/configuration.rst]: => `defines`

File: bundles/routing_auto/configuration.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+**type**: ``boolean`` **default**: ``true``
+
+Look for the configuration file ``cmf_routing_auto.yml`` in `Resource/config` folder of all
+available bundles.
+
+``persistence``
+~~~~~~~~~~~~~~~
+
+``phpcr``
+.........
```
Comment (dantleech) [inline bundles/routing_auto/configuration.rst]: Work in progress?

File: bundles/routing_auto/configuration.rst [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    .. code-block:: php
+
+        use Symfony\Component\DependencyInjection\Definition;
+
+        $definition = new Definition('AppBundle\RoutingAuto\CustomAdapter');
+        $definition->addTag('cmf_routing_auto.adapter', array('alias' => 'custom_adapter'));
+
+        $container->setDefinition('app.custom_adapter', $definition);
+
```
Comment (dantleech) [inline bundles/routing_auto/configuration.rst]: I don't think this DI configuration belongs here - shall we create a new document `Creating a custom adapter` with this information and then either link to it from here, or concentrate instead on that document and do this one later?

Comment (dbu) [inline bundles/routing_auto/configuration.rst]: i think this section would do a good separate chapter as it is now, just needs a title and an introduction that the bundle comes with phpcr support by default and if you need anything else you need a custom adapter.

File: bundles/routing_auto/configuration.rst (hunk 2 above)
Comment (dbu) [inline bundles/routing_auto/configuration.rst]: lets add a `.. todo` line maybe? @pcabreus seems to not use phpcr and we should know about it so better @dantleech or me document it in a separate pull request.

**Comment by pcabreus** [comment]
Thank you. The problem is I don't use ODMAdapter with the RoutingAuto, I've implemented my own ORMAdapter and I had to find a way to use it with RoutingAutoBundle. And I found out the bundle have a tag adapter parameter config, but it was not properly documented. That's why I propose this PR.

File: bundles/routing_auto/configuration.rst (hunk 2 above)
Comment (dantleech) [inline bundles/routing_auto/configuration.rst]: I think, as you say, we can turn this into its own chapter as it is now, and the `Configuration` chapter (which this PR proposes) can be added as a separate addition later. What do you think @pcabreus

File: book/routing.rst [hunk 4]
```diff
@@ -521,6 +521,145 @@ The PHPCR-ODM routes support more things, for example route parameters,
 requirements and defaults. This is explained in the
 :ref:`route document section in the RoutingBundle documentation <bundle-routing-document>`.
 
+The ORM Route Entity
```
Comment (dbu) [inline book/routing.rst]: very cool to have a good orm example. i wonder if it fits into the "book" section like this. the rest of the "book" is all about the phpcr-odm storage. i would see this as a chapter in the [routing bundle](http://symfony.com/doc/current/cmf/bundles/routing/index.html) documentation. and we can link to that chapter from here as well as from the [dynamic router orm section](http://symfony.com/doc/current/cmf/bundles/routing/dynamic.html#doctrine-orm-integration). wdyt?

File: book/routing.rst [hunk 5]
```diff
@@ -521,6 +521,145 @@ The PHPCR-ODM routes support more things, for example route parameters,
...
+--------------------
+
+The example in this section applies if you use the ORM route provider
+(``Symfony\Cmf\Bundle\RoutingBundle\Doctrine\Orm\RouteProvider``). It uses the
+``staticPrefix`` field of the
+``Symfony\Cmf\Bundle\RoutingBundle\Doctrine\Orm\Route`` to match routes.
+
+Symfony-Cmf routing system allows us loading whatever content from a route.
```
Comment (dbu) [inline book/routing.rst]: we usually write this as "Symfony CMF"

File: book/routing.rst [hunk 6]
```diff
@@ -521,6 +521,145 @@ The PHPCR-ODM routes support more things, for example route parameters,
...
+That means an entity route can reference to several entities. But Doctrine ORM
```
Comment (dbu) [inline book/routing.rst]: i would say "... to different types of entities." one route only points to a specific entity, not to many.

File: bundles/routing_auto/adapter.rst [hunk 7]
```diff
@@ -0,0 +1,80 @@
+.. index::
+    single: Adapter; RoutingAutoBundle
+
+Adapter
+=======
+
+Adapters will abstract all persistence operations. PhpcrOdmAdapter is available
+to be use it with Doctrine PHPCR-ODM persistence layer by default. If you want
+to use another persistence layer, you should implement your own adapter.
```
Comment (dbu) [inline bundles/routing_auto/adapter.rst]: Adapters abstract persistence operations. The ``PhpcrOdmAdapter`` is available for the Doctrine PHPCR-ODM persistence layer. If you want to use a different persistence layer, you need to implement your own adapter.

File: bundles/routing_auto/adapter.rst [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+            >
+                <tag name="cmf_routing_auto.adapter" alias="custom_adapter"/>
+            </service>
+        </container>
+
+    .. code-block:: php
+
+        use Symfony\Component\DependencyInjection\Definition;
+
+        $definition = new Definition('AppBundle\RoutingAuto\CustomAdapter');
```
Comment (dbu) [inline bundles/routing_auto/adapter.rst]: please use CustomAdapter::class and a `use` statement for the class.

File: bundles/routing_auto/adapter.rst [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+    .. code-block:: yaml
+
+        # app/config/config.yml
+        cmf_routing_auto:
+            adapter: custom_adapter
+
+    .. code-block:: xml
+
+        .. todo
```
Comment (dbu) [inline bundles/routing_auto/adapter.rst]: can you find out this one? best take another xml configuration as pattern and adjust it for this

**Comment by dbu** [comment]
thanks a lot! i did some minor cleanup and squashed the commits, then merged as e24c63f484172e041a9902a5e01553e31187f6d1

**Pull request closed without merging**